PlayApiとの接続処理の作成

yasu307 opened the pull request:
## 変更内容
[d8498d3](https://github.com/yasu307/todo-app-angular/pull/8/commits/d8498d3c43e0ac4af95fe8541db8741137f6c142)
- Backendの接続先をInMemoryDataからPlayApiに変更

[77b4149](https://github.com/yasu307/todo-app-angular/pull/8/commits/77b4149d783466f3550faf0e0d0e66c5fe6971b7)
- PlayApiではEnumのcodeが0から始まっていたので、Angularでもそれに合わせた

[1a864a7](https://github.com/yasu307/todo-app-angular/pull/8/commits/1a864a75db2d84e692399c49aba374265383a7a1)
- todo/categoryを追加するApiの返り値が追加したデータのidなので、それに合わせてAngularのメソッドの変更
- Apiから帰ってくるエラーに合わせてエラー処理を変更

## 画面変更
なし

yasu307 committed:
change backend setting

yasu307 committed:
change code of State/Color Options to match Play Api

yasu307 committed:
change addTodo/addCategory method to match Play Api

h-oikawa-jp commented on src/app/views/todo/todo.service.ts:
```diff
@@ -10,7 +10,7 @@ export class TodoService {
   private allTodoSource = new ReplaySubject<Todo[]>(1)
 
-  private todosUrl = "api/todos"
+  private todosUrl = "http://localhost:9000/api/todos"
```
APIのルートURL(ドメイン)などは検証環境や本番など環境毎に変更する事が多いので、
余裕があればコードにベタ書きではなく設定ファイルや環境変数などで容易に切り替えられるようにしてみてください。

yasu307 replied:
@h-oikawa-jp 
[e17eb84](https://github.com/yasu307/todo-app-angular/pull/8/commits/e17eb848ca688da5b067880368d82af38bbd4f32)にて修正しました。
修正内容は以下です。
- localデータを用いる環境ファイルを作成し、in-memory-dataの設定もこちらに移動
- 環境ファイルにバックエンドApiのURLを格納するように変更
- 本番用の環境ファイルもデフォルトと同じ内容で作成

ご確認よろしくお願いいたします。

yasu307 committed:
Change url that update method request to

yasu307 committed:
Change environment files

localの環境ファイルを作成し、in-memory-dataの設定をこちらに移動
環境ファイルにバックエンドApiのURLを格納するように変更
本番用の環境ファイルもデフォルトと同じ内容で作成

yasu307 commented:
[d55caf8](https://github.com/yasu307/todo-app-angular/pull/8/commits/d55caf838e613f8f0037e0ce323cdba072b1a119)にてPlayApiのroutes変更に合わせて、リクエスト先のURLを変更しました。ご確認よろしくお願いいたします。

h-oikawa-jp commented:
![LGTM](https://image.lgtmoon.dev/171862)
